crates/api/api_utils/src/notify.rs

```diff
@@ -272,6 +272,57 @@ pub async fn notify_private_message(
   Ok(())
 }
 
+/*
+TODO
+
+The following actions need to notify (both from api and federation):
+ModRemoveComment
+ModRemoveCommunity -> actually an admin action, needs rename. should notify all community mods?
+ModRemovePost
+ModBan -> ban user from site, needs rename
+ModBanFromCommunity
+ModLockPost
+ModAdd -> add instance admin, needs rename
+ModAddCommunity
+ModFeaturePost
+ModChangeCommunityVisibility -> should this notify and if so who?
+ModTransferCommunity -> notify both old and new top mods?
+
+Purges should probably be secret:
+- AdminPurgeComment
+- AdminPurgeCommunity
+- AdminPurgePerson
+- AdminPurgePost
+
+These dont affect any specific user and dont need to notify:
+- AdminAllowInstance
+- AdminBlockInstance
+*/
+pub fn notify_mod_action<T>(action: T, context: &LemmyContext)
+where
+  T: ModActionNotify + Send + 'static,
+{
+  let context = context.clone();
+  spawn_try_task(async move {
+    let Ok(local_recipient) =
+      LocalUserView::read_person(&mut context.pool(), action.target_person_id()).await
+    else {
+      return Ok(());
+    };
+
+    // TODO: Is there any good way to get the ModlogCombinedId here? Otherwise we need to get rid
+    //       of triggers and insert to ModlogCombined from Rust. Or otherwise the `notification`
+    //       table would need a separate column for each possible mod action.
```

Comment:
I think this is unavoidable: the notifications table will need new id columns for all the mod actions done to a person. 

The modlog_combined table shouldn't be used for this. The mod action endpoints (and the mod actions federation receives) will need to create notification rows / do the notifying for local users.

Reply:
I suppose this makes sense, then we treat modlog_combined as an implementation detail which is not public and may change in the future